### PR TITLE
add load balancer health check

### PR DESCRIPTION
**Pull request by qiutingjun**
- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features

add health check for load balancer

[[ovn-nb document](https://www.ovn.org/support/dist-docs/ovn-nb.5.html)](https://www.ovn.org/support/dist-docs/ovn-nb.5.html)

``` bash
     Health Checks:

       OVN  supports  health checks for load balancer endpoints, for IPv4 load
       balancers only. When health checks are enabled, the load balancer  uses
       only healthy endpoints.

       Suppose      that      vips      contains      a     key-value     pair
       10.0.0.10:80=10.0.0.4:8080,20.0.0.4:8080. 
       To enable health  checks  for this  virtual’s endpoints, 
       add two key-value pairs to ip_port_mappings,
       with keys 10.0.0.4 and 20.0.0.4, 
       and add to health_check a reference to
       a Load_Balancer_Health_Check row whose vip is set to 10.0.0.10.

       health_check: set of Load_Balancer_Health_Checks
              Load balancer health checks associated with this load balancer.

       ip_port_mappings: map of string-string pairs
              Maps  from endpoint IP to a colon-separated pair of logical port
              name and source IP, e.g. port_name:sourc_ip. Health  checks  are
              sent to this port with the specified source IP.

              For  example, in the example above, IP to port mappings might be
              defined         as         10.0.0.4=sw0-p1:10.0.0.2          and
              20.0.0.4=sw1-p1:20.0.0.2,  if  the  values  given  were suitable
              ports and IP addresses.


```

```bash
uuid=`ovn-nbctl --bare --columns _uuid find load_balancer name=test-vpc3-tcp-lb0`

# 关联 ip_port_mappings 到loadbalancer

ovn-nbctl --wait=sb set load_balancer $uuid ip_port_mappings:192.168.4.4=vpc3-busybox03.vpc3:192.168.4.14

ovn-nbctl --wait=sb set load_balancer $uuid ip_port_mappings:192.168.4.5=vpc3-busybox04.vpc3:192.168.4.15

uuid3=`ovn-nbctl --id=@hc create Load_Balancer_Health_Check vip="10.120.31.215\:8000" -- add Load_Balancer $uuid health_check @hc`

# echo $uuid3
66117312-2b7a-4920-b903-fadb3a842b3b

ovn-nbctl set Load_Balancer_Health_Check $uuid3 options:interval=5 options:timeout=20 options:success_count=3 options:failure_count=3



# ovn-nbctl list Load_Balancer_Health_Check
_uuid               : 66117312-2b7a-4920-b903-fadb3a842b3b
external_ids        : {}
options             : {failure_count="3", interval="5", success_count="3", timeout="20"}
vip                 : "10.120.31.215:8000"

```

### Which issue(s) this PR fixes:
Fixes #(issue-number)

### <samp>🤖 Generated by Copilot at c66e373</samp>

This pull request adds support for ip port mappings and health checks to load balancers for services and endpoints in the `kube-ovn` controller. It also improves the readability, consistency, and error handling of the code in various files, updates the module versions in `go.mod`, and adds unit tests for the new functionality. Additionally, it renames some files to follow the Go naming convention and avoid linting errors.

### <samp>🤖 Generated by Copilot at c66e373</samp>

> _We're the crew of the OVN ship, we code with skill and pride_
> _We manage load balancers and health checks, we keep the service alive_
> _We follow Go conventions and we test our code with care_
> _We heave away on the count of three, we're the best in the open air_

### <samp>🤖 Generated by Copilot at c66e373</samp>

*  Rename files to follow Go naming convention ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-945b73ad15889230d50be6f6fbd9e7352e96338574ce6f8a53b7a32b2e334739), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-cba763fee4fe58f36170742cfd47077f68084251db8632bcadbcf13fd103d9b8), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-5c7a894955e2bc94f38a61b0b223a15b3a573205553d2b4991fa0ce3bce7b087), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-568a80c359df489e53258425df044c8f9cb75fed197e579f0d5769f93008f1e7))
*  Update dependencies in `go.mod` file ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L39-R39), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L232-R239))
*  Add new value to exclude list of gosec tool in `Makefile` ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L828-R828))
*  Add and modify mock interfaces and methods for the ovs package in `mocks/pkg/ovs/interface.go` ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71R952-R979), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71L971-R1030), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71L980-R1038), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71R1056-R1069), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71R1084-R1208), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71R3224-R3251), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71L3048-R3302), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71L3057-R3310), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-2eadf7fb3dd75cdfd04457a9b7e244a2bfb9aa9bb790ff6ed89b86599c0a4c71R3328-R3341))
*  Improve readability and consistency of code by declaring variables in single var blocks, simplifying if statements, removing unused code, and renaming variables in various files ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L19-R23), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L40-R47), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L62-R76), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L71-R83), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L77-R89), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L115-R136), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L128), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L180-R197), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L192-R238), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L216-R254), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528R268-R271), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L242), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L268-R303), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L255-R255), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L385-R385), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L430), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L444-R452), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L488-R551), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89L541-R564), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L25-R29), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L31-R35), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L37-R51), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L202-R215), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L213-R221), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988L378-R386), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L24-R28), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L35-R42), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L67-R77), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L458-R472), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L466-R485), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L496-R517), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L504-R531), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L529), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L624-R669), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L250-R255), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L262-R271), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L352-R365), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L366-R379), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L381-R394), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L401-R414), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-1bbd8fc68082905bd8c8204a29a746985193081fb932bedaa86f4f25abc2dc02L333-R340), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-c53f89147c39d7d4c3d3046d288a0cea09719776e947bfb3ef01ea29f75b1fb0L562-R562), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-50fe35ba11ad04edb53bfaccef915df3650a81844495ed73d3c158e975c04c49L83-R88), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-50fe35ba11ad04edb53bfaccef915df3650a81844495ed73d3c158e975c04c49R96-R105))
*  Support new functionality of adding ip port mappings and health checks to load balancers by modifying the logic of deleting or adding vips, adding new parameters and return types to existing functions, and adding new fields to existing structs in various files ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L192-R238), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L216-R254), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528R268-R271), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L268-R303), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L582-R622), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-7fa512aca42bc3bdafdaed430f6c2000a24cdf0973b09a703a88c7a9b7b054d1L603-R642), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L262-R271), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L275-R283), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L366-R379), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L381-R394), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L401-R414), [link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-1bbd8fc68082905bd8c8204a29a746985193081fb932bedaa86f4f25abc2dc02L333-R340))
*  Add new file (`pkg/ovs/ovn-nb-load_balancer_health_check.go`) that implements the LoadBalancerHealthCheck interface and provides the methods for managing load balancer health checks in the OVN NB database ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-b193db1a9429f799227f035b255c48693043e1807435a181bae5f5f58c05aadaR1-R288))
*  Add new file (`pkg/ovs/ovn-nb-load_balancer_health_check_test.go`) that provides unit tests for the LoadBalancerHealthCheck interface and its methods ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-bfd041dba2e9723d827d8cb8319204ff483edd6d1489966769e2718c61202453R1-R337))
*  Add new log message to the AddNat function in `pkg/ovs/ovn-nb-nat.go` to improve error handling and debugging ([link](https://github.com/kubeovn/kube-ovn/pull/3216/files?diff=unified&w=0#diff-1a843dfb6cc5e7791427ee159e4f1243b5fa6db07b5fefdbafaed21ebfe6692bR31))

### CODE REVIEW

**Review by bobz965 — CHANGES_REQUESTED** [review]

**Comment by bobz965** [comment]
lb 健康检查:

- 仅 switch lb rule 支持健康检查
- 仅当存在多个后端（ip port map）时，支持添加健康检查
- slr e2e 添加健康检查的 e2e 校验 service monitor 的状态

File: pkg/controller/endpoint.go [hunk 1]
```diff
@@ -189,37 +225,98 @@ func (c *Controller) handleUpdateEndpoint(key string) error {
...
+										Subnet: subnetName,
+									},
+								},
+								metav1.CreateOptions{},
+							); err != nil {
+							klog.Errorf("failed to create health check vip from %s %s, %v", vpcName, subnetName, err)
+							return err
+						}
+					} else {
+						klog.Errorf("failed to get health check vip from %s %s, %v", vpcName, subnetName, err)
```
Comment (bobz965) [inline pkg/controller/endpoint.go:257]:
```suggestion
						klog.Errorf("failed to get health check vip from vpc %s subnet %s, %v", vpcName, subnetName, err)
```

File: pkg/controller/endpoint.go [hunk 2]
```diff
@@ -189,37 +225,98 @@ func (c *Controller) handleUpdateEndpoint(key string) error {
...
+				}
+
+				switch util.CheckProtocol(lbVip) {
+				case kubeovnv1.ProtocolIPv4:
+					checkIP = checkVip.Status.V4ip
+				case kubeovnv1.ProtocolIPv6:
+					checkIP = checkVip.Status.V6ip
+				}
+				if checkIP == "" {
+					err = fmt.Errorf("failed to get health check vip from %s %s", vpcName, subnetName)
```
Comment (bobz965) [inline pkg/controller/endpoint.go:273]:
```suggestion
					err = fmt.Errorf("failed to get health check vip from vpc %s subnet %s", vpcName, subnetName)
```

File: pkg/controller/gc.go [hunk 3]
```diff
@@ -485,66 +488,80 @@ func (c *Controller) gcLoadBalancer() error {
...
+			err error
+		)
+
+		if lb, err = c.OVNNbClient.GetLoadBalancer(lbName, true); err != nil {
+			klog.Errorf("get LB %s: %v", lbName, err)
+			return err
+		}
+
+		if lb == nil {
+			klog.Infof("load balancer %q not found", lbName)
```
Comment (bobz965) [inline pkg/controller/gc.go:514]:
```suggestion
			klog.Infof("load balancer %q already deleted", lbName)
```

File: pkg/controller/subnet.go [hunk 4]
```diff
@@ -235,6 +236,10 @@ func (c *Controller) processNextDeleteSubnetWorkItem() bool {
 		return false
 	}
 
+	log.Println("======================================================")
```
Comment (bobz965) [inline pkg/controller/subnet.go:239]: remove debug log

File: pkg/controller/subnet.go [hunk 5]
```diff
@@ -941,6 +946,15 @@ func (c *Controller) handleDeleteSubnet(subnet *kubeovnv1.Subnet) error {
 		return err
 	}
 
+	// if err := c.config.KubeOvnClient.
```
Comment (bobz965) [inline pkg/controller/subnet.go:949]: remove not used code

File: pkg/controller/switch_lb_rule.go [hunk 6]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
 		return err
 	}
+
+	lbhcs, err := c.OVNNbClient.ListLoadBalancerHealthChecks(
+		func(lbhc *ovnnb.LoadBalancerHealthCheck) bool {
+			return util.ContainsString(info.Vips, lbhc.Vip)
+		},
+	)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		klog.Errorf("failed to list load balancer health checks %s,err: %v", info.Vips, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:267]:
```suggestion
		klog.Errorf("failed to list load balancer health checks matched vips %s,err: %v", info.Vips, err)
```

File: pkg/controller/switch_lb_rule.go [hunk 7]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+	vips := make(map[string]struct{})
+
+	for _, lbhc := range lbhcs {
+		lbs, err := c.OVNNbClient.ListLoadBalancers(
+			func(lb *ovnnb.LoadBalancer) bool {
+				return util.ContainsString(lb.HealthCheck, lbhc.UUID)
+			},
+		)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			klog.Errorf("failed to list load balancer %s,err: %v", lbhc.Vip, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:280]:
```suggestion
			klog.Errorf("failed to list load balancer matched vip %s, err: %v", lbhc.Vip, err)
```

File: pkg/controller/switch_lb_rule.go [hunk 8]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+			return err
+		}
+
+		for _, lb := range lbs {
+			err = c.OVNNbClient.LoadBalancerDeleteHealthCheck(lb.Name, lbhc.UUID)
+			if err != nil && !k8serrors.IsNotFound(err) {
+				klog.Errorf("failed to delete load balancer health checks health checks %s from load balancer %s ,err: %v", lbhc.Vip, lb.Name, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:287]:
```suggestion
				klog.Errorf("failed to delete load balancer health checks health checks matched vip %s from load balancer %s, err: %v", lbhc.Vip, lb.Name, err)
```

File: pkg/controller/switch_lb_rule.go [hunk 9]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+				return err
+			}
+
+			err = c.OVNNbClient.LoadBalancerDeleteIPPortMapping(lb.Name, lbhc.Vip)
+			if err != nil && !k8serrors.IsNotFound(err) {
+				klog.Errorf("failed to delete ip port mappings %s from load balancer %s ,err: %v", lbhc.Vip, lb.Name, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:293]: format

File: pkg/controller/switch_lb_rule.go [hunk 10]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+		}
+	}
+
+	err = c.OVNNbClient.DeleteLoadBalancerHealthChecks(
+		func(lbhc *ovnnb.LoadBalancerHealthCheck) bool {
+			return util.ContainsString(info.Vips, lbhc.Vip)
+		},
+	)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		klog.Errorf("delete load balancer health checks %s,err: %v", info.Vips, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:309]: format

File: pkg/controller/switch_lb_rule.go [hunk 11]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+	}
+
+	for vip := range vips {
+		lbhcs, err = c.OVNNbClient.ListLoadBalancerHealthChecks(
+			func(lbhc *ovnnb.LoadBalancerHealthCheck) bool {
+				return lbhc.ExternalIDs[util.SwitchLBRuleSubnet] == vip
+			},
+		)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			klog.Errorf("failed to list load balancer ,err: %v", err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:320]: format

File: pkg/controller/switch_lb_rule.go [hunk 12]
```diff
@@ -234,6 +257,79 @@ func (c *Controller) handleDelSwitchLBRule(info *SlrInfo) error {
...
+			return err
+		}
+
+		if len(lbhcs) == 0 {
+			err = c.config.KubeOvnClient.KubeovnV1().Vips().Delete(context.Background(), vip, metav1.DeleteOptions{})
+			if err != nil {
+				klog.Errorf("failed to delete vip %s for load balancer health check ,err: %v", vip, err)
```
Comment (bobz965) [inline pkg/controller/switch_lb_rule.go:327]: format

**Comment by bobz965** [comment]
please attach the docs pr

**Review by bobz965 — APPROVED** [review]

**Comment by bobz965** [comment]
hi @oilbeater, could you help review?

**Comment by qiutingjun** [comment]
> please attach the docs pr

https://github.com/kubeovn/docs/pull/129

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: Makefile [hunk 13]
```diff
@@ -910,7 +910,7 @@ lint:
 		echo "Code differs from gofmt's style" 1>&2 && exit 1; \
 	fi
 	@GOOS=linux go vet ./...
-	@GOOS=linux gosec -exclude=G204,G306,G404,G601,G301 -exclude-dir=test -exclude-dir=pkg/client ./...
+	@GOOS=linux gosec -exclude=G101,G204,G306,G404,G601,G602,G301 -exclude-dir=test -exclude-dir=pkg/client ./..
```
[the remark this answers is not in the document]
Comment (qiutingjun) [reply Makefile]: revert done

**Review by oilbeater — APPROVED** [review]

**Pull request merged**